RobotNavigation/Maze.cs

```diff
@@ -58,5 +63,17 @@ public void setCell(int x, int y, Cell c)
         {
             _maze[x, y] = c;
         }
+
+        public void setTargets(int x, int y)
+        {
+            _pos.X = x;
+            _pos.Y = y;
+            _posArray.Add(_pos);
+        }
+
+        public ArrayList getTargets()
```

Comment:
Do you know if you need this yet?  Why do you want to return the full list of targets on a Maze?

Reply:
So that you can loop through them to see if the maze has been solved

Reply:
You're jumping ahead here.  I would imagine the robot moves from Position to Position via the _maze array object.  At each one you're checking if it's a target or not by checking the Type (wall, space, target).  No need to loop through a list of target positions when the Maze contains the full array of positions.